### ISSUE TITLE
_goldens_web.dart error in web app only

### ISSUE THREAD

**Issue by joezhang** [firebase/flutterfire#10529]
Launching lib/main.dart on Chrome in debug mode...
main.dart:1
: Error: A non-null value must be returned since the return type 'bool' doesn't allow null.
_goldens_web.dart:59
  Future<bool> compare(double width, double height, Uri golden) async {
               ^
Failed to compile application.
Exited

**Comment by darshankawar** [comment]
@joezhang 
When you open a new issue, please try to use [new issue template](https://github.com/firebase/flutterfire/issues/new?assignees=&labels=Needs+Attention%2C+type%3A+bug&template=---bug-report.md&title=%F0%9F%90%9B+%5BPLUGIN_NAME_HERE%5D+Your+issue+title+here) and fill out requested details so that we'll have enough info to work with. 

For now, please provide `flutter doctor -v`, complete minimal reproducible code sample along with entire stacktrace.

**Comment by joezhang** [comment]
flutter doctor -v
[✓] Flutter (Channel stable, 3.7.5, on macOS 13.2.1 22D68 darwin-arm64, locale en-CA)
    • Flutter version 3.7.5 on channel stable at /Users/zhongyuzhang/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c07f788888 (5 days ago), 2023-02-22 17:52:33 -0600
    • Engine revision 0f359063c4
    • Dart version 2.19.2
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/zhongyuzhang/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Community Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.75.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.2.1 22D68 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**Comment by joezhang** [comment]
~% flutter --version
Flutter 3.7.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision c07f788888 (5 days ago) • 2023-02-22 17:52:33 -0600
Engine • revision 0f359063c4
Tools • Dart 2.19.2 • DevTools 2.20.1
~% flutterfire --version
0.2.7

**Comment by darshankawar** [comment]
@joezhang 
Can you run the same scenario using latest master to see if the error persists on it a well ?

`flutter channel master`
`flutter upgrade`

**Comment by joezhang** [comment]
~/Documents/Flutter/testapp % flutter --version
Flutter 3.9.0-1.0.pre.6 • channel master • https://github.com/flutter/flutter.git
Framework • revision b85f195d39 (31 minutes ago) • 2023-03-01 18:37:53 -0500
Engine • revision 03c9a9fc8d
Tools • Dart 3.0.0 (build 3.0.0-285.0.dev) • DevTools 2.22.2
----------------------------------------------------------------------------------------------~/Documents/Flutter/testapp % flutter doctor -v
[✓] Flutter (Channel master, 3.9.0-1.0.pre.6, on macOS 13.2.1 22D68 darwin-arm64, locale en-CA)
    • Flutter version 3.9.0-1.0.pre.6 on channel master at /Users/zhongyuzhang/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b85f195d39 (31 minutes ago), 2023-03-01 18:37:53 -0500
    • Engine revision 03c9a9fc8d
    • Dart version 3.0.0 (build 3.0.0-285.0.dev)
    • DevTools version 2.22.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/zhongyuzhang/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Community Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.75.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.2.1 22D68 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

• No issues found!
----------------------------------------------------------------------------------------------
~/Documents/Flutter/testapp % flutter pub add firebase_core
Resolving dependencies... 
+ firebase_core 2.7.0
+ firebase_core_platform_interface 4.5.3
+ firebase_core_web 2.2.1
+ flutter_web_plugins 0.0.0 from sdk flutter
+ plugin_platform_interface 2.1.4
Changed 5 dependencies!
----------------------------------------------------------------------------------------------
~/Documents/Flutter/testapp % flutterfire configure
i Found 2 Firebase projects.                                                                                                         
✔ Select a Firebase project to configure your Flutter application with · testapp-f19a2 (testapp)                                     
✔ Which platforms should your configuration support (use arrow keys & space to select)? · android, ios, macos, web                   
i Firebase android app com.example.testapp is not registered on Firebase project testapp-f19a2.                                      
i Registered a new Firebase android app on Firebase project testapp-f19a2.                                                           
i Firebase ios app com.example.testapp is not registered on Firebase project testapp-f19a2.                                          
i Registered a new Firebase ios app on Firebase project testapp-f19a2.                                                               
i Firebase macos app com.example.testapp.RunnerTests is not registered on Firebase project testapp-f19a2.                            
i Registered a new Firebase macos app on Firebase project testapp-f19a2.                                                             
i Firebase web app testapp (web) is not registered on Firebase project testapp-f19a2.                                                
i Registered a new Firebase web app on Firebase project testapp-f19a2.                                                               
? The files android/build.gradle & android/app/build.gradle will be updated to apply Firebase configuration and gradle build plugins.
? The files android/build.gradle & android/app/build.gradle will be updated to apply Firebase configuration and gradle build plugins.
✔ The files android/build.gradle & android/app/build.gradle will be updated to apply Firebase configuration and gradle build plugins. Do you want to continue? · yes 

Firebase configuration file lib/firebase_options.dart generated successfully with the following Firebase apps:

Platform  Firebase App Id
web       1:273237930182:web:d7173b1b4a8d995b9ae7c7
android   1:273237930182:android:665a901f50c740c19ae7c7
ios       1:273237930182:ios:57d21db6290036349ae7c7
macos     1:273237930182:ios:61dfe973b3299dc49ae7c7
----------------------------------------------------------------------------------------------
~/Documents/Flutter/testapp % flutter run
Multiple devices found:
macOS (desktop) • macos  • darwin-arm64   • macOS 13.2.1 22D68 darwin-arm64
Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119
[1]: macOS (macos)
[2]: Chrome (chrome)
Please choose one (or "q" to quit): 2
Launching lib/main.dart on Chrome in debug mode...
../../../development/flutter/packages/flutter_test/lib/src/_goldens_web.dart:59:16: Error: A non-null value must be returned since
the return type 'bool' doesn't allow null.
  Future<bool> compare(double width, double height, Uri golden) async {
               ^
Waiting for connection from debug service on Chrome...             10.4s
Failed to compile application.

**Comment by joezhang** [comment]
same error, please check the step-by-step above to reproduce the issue.

**Comment by darshankawar** [comment]
Thanks for the update and patience. I followed all the steps to create a new flutter project, configured firebase app using `flutterfire configure`, added `firebase_core` and ran on web (flutter run -d chrome) which compiled and ran properly without any issues:

<img width="1197" alt="Screenshot 2023-03-02 at 5 33 49 PM" src="https://user-images.githubusercontent.com/67046386/222423978-8735af64-1b32-4521-8c6f-f9efdbbf22a3.png">

```

Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             23.1s
This app is linked to the debug service: ws://127.0.0.1:53550/OfZeTcM5_2o=/ws
Debug service listening on ws://127.0.0.1:53550/OfZeTcM5_2o=/ws
```

> * The file _goldens_web.dart is generated;

I did not see this file getting generated. Where is this ideally located ?

**Comment by joezhang** [comment]
The file _goldens_web.dart is in flutter>packages>flutter_test>lib>src

**Comment by joezhang** [comment]
~/development/flutter/packages/flutter_test/lib/src % ls
_binding_io.dart			image.dart
_binding_web.dart			matchers.dart
_goldens_io.dart			nonconst.dart
**_goldens_web.dart**			platform.dart
_matchers_io.dart			restoration.dart
_matchers_web.dart			stack_manipulation.dart
accessibility.dart			test_async_utils.dart
all_elements.dart			test_compat.dart
animation_sheet.dart			test_default_binary_messenger.dart
binding.dart				test_exception_reporter.dart
buffer_matcher.dart			test_pointer.dart
controller.dart				test_text_input.dart
deprecated.dart				test_text_input_key_handler.dart
event_simulation.dart			test_vsync.dart
finders.dart				widget_tester.dart
frame_timing_summarizer.dart		window.dart
goldens.dart

**Comment by darshankawar** [comment]
I do see the said file:

<img width="541" alt="Screenshot 2023-03-03 at 5 04 27 PM" src="https://user-images.githubusercontent.com/67046386/222710331-f1cdbd0f-d230-48c4-8104-cb1ea5fc9434.png">

But I am unable to replicate the error per my earlier [verification](https://github.com/firebase/flutterfire/issues/10529#issuecomment-1451762759).

Based on the report, keeping this issue open and labeling for team's attention.

**Comment by joezhang** [comment]
You didi not reproduce all the steps. All the steps you did below will not have any issue.  
_--"Thanks for the update and patience. I followed all the steps to create a new flutter project, configured firebase app using flutterfire configure, added firebase_core and ran on web (flutter run -d chrome) which compiled and ran properly without any issues:"_

The web issue will happen only after the step "**flutterfire configure**"

Please try again.
Thanks.

**Comment by Lyokone** [comment]
@joezhang Since we are unable to reproduce this with steps, please provide a repository that will help us troubleshoot your issue.
Thanks

**Comment by joezhang** [comment]
it seems the issue had been fixed with the flutter upgrade recently, thanks.
